web/src/compose_closed_ui.js

```diff
@@ -81,32 +81,45 @@ function update_reply_button_state(disable = false) {
     }
 }
 
-function update_new_conversation_button(btn_text) {
-    $("#new_conversation_button").text(btn_text);
+function update_new_conversation_button(btn_text, is_direct_message_narrow) {
+    const $new_conversation_button = $("#new_conversation_button");
+    $new_conversation_button.text(btn_text);
+    // In a direct-message narrow, the new conversation button should act
+    // like a new direct message button
+    if (is_direct_message_narrow) {
+        $new_conversation_button.addClass("compose_new_direct_message_button");
+        $new_conversation_button.removeClass("compose_new_conversation_button");
+    } else {
+        // Restore the usual new conversation button class, if it was
+        // changed after a previous direct-message narrow visit
+        $new_conversation_button.addClass("compose_new_conversation_button");
+        $new_conversation_button.removeClass("compose_new_direct_message_button");
+    }
 }
 
 function update_new_direct_message_button(btn_text) {
     $("#new_direct_message_button").text(btn_text);
 }
 
-function update_buttons(text_stream, disable_reply) {
+function update_buttons(text_stream, is_direct_message_narrow, disable_reply) {
```

Comment:
I might consider adjusting the other calls to pass an explicit `false` for this parameter for readability; or having explicit  `is_direct_message_narrow = false` defaults -- though I guess we'll fix this when we move this file to TypeScript. Generally hate having parameters with a `true | implicit undefined` type.

Reply:
Yeah :) I felt a distinct "When in Rome..." to handle it that way, but I agree that a more explicit style is always better.